### DISCUSSION TITLE
About the hybrid keyboard for PC and Mac

### DISCUSSION THREAD

**Discussion by louies0623** [helloSystem/hello#399] [Show and tell]
![818t3rhOubL _AC_SX569_ (1)](https://user-images.githubusercontent.com/44593430/201001449-b8f47d42-39dc-45cd-ad10-c6100cfa8f29.jpg)
I have previously responded that there is a problem with the placement of the PC control keys, which is more pronounced on this keyboard.  This is a Logitech K380, a hybrid keyboard for PC and Mac, but on Hello os, it is confusing to use due to the swapping of control keys in the system, you can buy it and try it.[Logitech K380](https://www.amazon.de/-/en/Logitech-Bluetooth-multi-device-easy-switch-shortcuts/dp/B013SL1YXO)

**Comment by probonopd** [comment]
This is the officially supported keyboard:
https://www.raspberrypi.com/products/raspberry-pi-keyboard-and-hub/

**Reply by louies0623** [reply]
Because some users have to switch between PC and Mac, they will use a mixed PC and MAC keyboard, and your replacement of the control keys will conflict with keyboards like this.

**Reply by probonopd** [reply]
On Apple keyboards, the keys are not reversed, so that Command still works as Command.

**Comment by louies0623** [comment]
By the way, your system key replacement is reversed on my macbook's keyboard.

**Reply by probonopd** [reply]
Please elaborate.

**Reply by louies0623** [reply]
The "command" and "option" of hello os are the opposite of the mac keyboard arrangement.

**Reply by louies0623** [reply]
Bug:https://youtu.be/yk9Mmcg93GU

**Reply by louies0623** [reply]
That's the problem that shouldn't happen.

**Reply by probonopd** [reply]
Shouldn't be like this. Is it a virtual machine?

**Reply by louies0623** [reply]
The video is a virtual machine because it is too difficult to boot on a physical machine, but it is the same on a physical machine.

**Reply by probonopd** [reply]
It's a bug then.

**Reply by probonopd** [reply]
Did you have any non-Apple keyboards attached when the physical machine booted?

**Reply by louies0623** [reply]
That's why I once suggested you remove the code for key substitution.

**Reply by louies0623** [reply]
I tried it, and the buttons are swapped.

**Reply by probonopd** [reply]
Possibly `/usr/local/etc/xdg/Xmodmap` needs to be deleted/edited?

**Reply by louies0623** [reply]
Otherwise, you can use an iso that deletes /usr/local/etc/xdg/Xmodmap to let me test, this might verify that it works.

**Reply by probonopd** [reply]
I don't understand the issue. Please boot with _only_ Apple keyboards attached to the system. No third-party keyboards. Does it work then?

helloSystem cannot support a mix of Apple keyboards and non-Apple keyboards built in/attached to the same computer.

**Reply by louies0623** [reply]
with only Apple keyboards attached to the system. No third-party keyboards. Does it work then?

**How do you know if you haven't do it!**

**Reply by probonopd** [reply]
When I was working on the keyboard stuff, of course it was working.

**Reply by louies0623** [reply]
Update: I was wrong, Sorry
The PC keyboard is arranged correctly. Switching to the Mac control key arrangement is problematic. The root problem is that "Alt as ⌘ button" ⌘ is connected to the Mac's Option key
https://youtu.be/rraqX3OdXic
